### PR TITLE
Disk Collection Browser

### PR DESCRIPTION
**Pull request by dcaputi1**
The Disk Collection Browser (aka Disk Browser), allows the user to conveniently load and unload floppy disk image files with the click of the mouse from one simple dialog box. This creates a nostalgic approximation of the floppy disk flip files where an Atari user would commonly keep their collection of disks.

For more details refer to Disk-Collection-Browser.rtf in the RespeQr project root folder.

### CODE REVIEW

File: include/miscutils.h [hunk 1]
```diff
@@ -33,6 +33,14 @@ class GzFile : public QFile {
   qint64 writeData(const char *data, qint64 maxSize) override;
 };
 
+inline Qt::CaseSensitivity osCaseSensitivity() {
+#if defined Q_OS_WIN
```
Comment (josch1710) [inline include/miscutils.h]: MacOS file systems are by default case insensitive. Please change that such that Linux is case sensitive and the rest not.

File: include/miscutils.h [hunk 2]
```diff
@@ -59,7 +67,20 @@ class FileTypes : public QObject {
   };
   static FileType getFileType(const QString &fileName);
   __attribute__((unused)) static QString getFileTypeName(FileType type);
+
+  static const QStringList& getDiskImageTypes()
+  {
+      static QStringList list
+      {
+          "*.xfd", "*.atr", "*.pro", "*.atx"
+  #if !defined Q_OS_WIN
```
Comment (josch1710) [inline include/miscutils.h]: The same as above in the method osCaseSensitivity

File: include/mainwindow.h [hunk 3]
```diff
@@ -195,12 +204,11 @@ private slots:
   void saveAsTriggered(char no);                           // MIA
   void revertTriggered(char no);                           // MIA
 
-
   void bootOptionTriggered();    //
   void toggleMiniModeTriggered();//
   void toggleShadeTriggered();   //
   void showLogWindowTriggered(); //
-
+  void diskBrowserTriggered();   // what are all these trailing comment markers for?
```
Comment (josch1710) [inline include/mainwindow.h:211]: I don't know. Perhaps a placeholder for adding functionality comments in far future.

File: src/mainwindow.cpp [hunk 4]
```diff
@@ -108,25 +113,34 @@ void MainWindow::logMessageOutput(QtMsgType type, const QMessageLogContext & /*c
...
+
   logFile->write(displayMsg);
   logFile->write("\n");
+
   if (type == QtFatalMsg) {
     logFile->close();
+#if defined(QT_NO_DEBUG) // allow debugger to survive fatal error
     abort();
+#elif defined (Q_OS_WIN) // TBD: non-windows? (QtCreator 10.x on Mojave is OK)
+    __debugbreak();
```
Comment (josch1710) [inline src/mainwindow.cpp:132]: Does __debugbreak work on Linux, Macos, and Windows?

**Review by josch1710 — CHANGES_REQUESTED** [review]
I made some comments. Please fix that.

**Review by josch1710 — CHANGES_REQUESTED** [review]
Please review my comments.

File: src/mainwindow.cpp (hunk 4 above)
Comment (dcaputi1) [reply src/mainwindow.cpp:132]: It worked on my intel based MacBook but not sure that means all MacOS. First hit on stack overflow has an incomplete solution: https://stackoverflow.com/questions/173618/is-there-a-portable-equivalent-to-debugbreak-debugbreak
... not sure where to go with this.

**Review by josch1710 — APPROVED** [review]

**Pull request merged**